SalesforceCommerceCloud/pwa-kit
# @W-16600055 Page Designer Banner displayed based on shopper context

# Description

GUS: [W-16600055](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE0000201drTYAQ/view)

Implements an example of how Shopper context can drive personalization for content delivered to the site via our first-party CMS (Page Designer) in the Retail React App

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- render a `instagram-promo-banner` page designer page in the product-list page
- update image-with-text.js not to add a LinkWrapper if link is not present

# How to Test-Drive This PR

- latest changes have been deployed to: https://cloud-jinsu-production2.mobify-storefront-staging.com/
- navigate to a PLP page: https://cloud-jinsu-production2.mobify-storefront-staging.com/global/en-GB/category/mens-clothing-jackets
- Verify no Instagram promo banner is displayed
- add the query param `?sourceCode=instagram` to update the shopper context: https://cloud-jinsu-production2.mobify-storefront-staging.com/global/en-GB/category/mens-clothing-jackets?sourceCode=instagram
- Verify an Instagram promo banner is displayed
- Resize the browser to the smallest width, verify the sizing of the banner and text displays in a visually appealing way
- Open the dev console and change the browser to mobile view, verify the mobile version of the banner is displayed
- set the query param `?sourceCode=fb` to update the shopper context: https://cloud-jinsu-production2.mobify-storefront-staging.com/global/en-GB/category/mens-clothing-jackets?sourceCode=fb
- Verify the Instagram promo banner is not displayed

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
